internal/endpoints/awsrulesfn/host.go

```diff
@@ -0,0 +1,69 @@
+package awsrulesfn
+
+import (
+	"fmt"
+	"net/netip"
+	"strings"
+	smithyerrep "github.com/aws/smithy-go/error/endpoints"
+	smithyhttp "github.com/aws/smithy-go/transport/http"
+
+)
+
+// IsVirtualHostableS3Bucket returns if the input is a DNS compatible bucket
+// name and can be used with Amazon S3 virtual hosted style addressing. Similar
+// to [rulesfn.IsValidHostLabel] with the added restriction that the length of label
+// must be [3:63] characters long, all lowercase, and not formatted as an IP
+// address.
+func IsVirtualHostableS3Bucket(input string, allowSubDomains bool, ec *smithyerrep.ErrorCollector) bool {
```

Comment:
We should reexamine this implementation in general, it should be greatly simplified once `IsValidHostLabel` in smithy-go becomes accessible here.

Reply:
hmm yah i agree that `IsVirtualHostableS3Bucket` shares some logic with `IsValidHostLabel` but it looks like the only thing that could potentially be removed is lines 55-63 (since that case is already checked in `IsValidHostLabel`). But the added benefit of having it also here also could be better error reporting, since it identifies that the invalid host occurred during the check of `IsVirtualHostableS3Bucket`.
The other three things that are checked for in `IsVirtualHostableS3Bucket`, which are not in `IsValidHostLabel` are
1. not IP address
2. not less 3 characters and not longer than 63 characters
3. no capital letters

So not sure how those could be simplified by utilizing `IsValidHostLabel`. Or is there something else that you are seeing that could simplify this?